libkbfs/ops.go

```diff
@@ -156,18 +211,33 @@ type createOp struct {
 	crSymPath string
 }
 
-func newCreateOp(name string, oldDir BlockPointer, t EntryType) *createOp {
+func newCreateOp(name string, oldDir BlockPointer, t EntryType) (*createOp, error) {
 	co := &createOp{
 		NewName: name,
 	}
-	co.Dir.Unref = oldDir
+	err := co.Dir.setUnref(oldDir)
+	if err != nil {
+		return nil, err
+	}
 	co.Type = t
-	return co
+	return co, nil
+}
+
+func newCreateOpForRootDir() *createOp {
+	return &createOp{
+		Type: Dir,
+	}
 }
 
 func (co *createOp) AddUpdate(oldPtr BlockPointer, newPtr BlockPointer) {
+	if co.Dir == (blockUpdate{}) {
+		panic("AddUpdate called on create op for root dir")
```

Comment:
Why would this indicate the root dir?  Couldn't it be due to any number of other bugs?